src/main/java/digital/slovensko/autogram/ui/gui/SigningDialogController.java

```diff
@@ -298,19 +309,20 @@ public void showHTMLVisualization(String html) {
         webViewContainer.setManaged(true);
     }
 
-    public void showPDFVisualization(String base64EncodedPdf) {
-        var engine = webView.getEngine();
-        engine.setJavaScriptEnabled(true);
-        engine.getLoadWorker().stateProperty().addListener((observable, oldState, newState) -> {
-            if (newState == Worker.State.SUCCEEDED) {
-                engine.executeScript(
-                        "displayPdf('" + base64EncodedPdf + "')");
-            }
+    public void showPDFVisualization(ArrayList<byte[]> data) {
+        data.forEach(page -> {
+            var imgView = new ImageView();
+            imgView.fitWidthProperty().bind(pdfVisualizationContainer.widthProperty().subtract(30));
```

Comment:
Toto tu bolo lebo som to s tym webview ani za svet nevedel dobre nastavit, ale native imgview by mohol ist ci?

Reply:
Kvôli tomuto scrollbaru som to offsetol. Ak necháme 0, vznikne aj horizontálny scrollbar.